### ISSUE TITLE
Classification model export2tflite model failed

### ISSUE THREAD

**Issue by jshh0401** [ultralytics/yolov5#13619]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

Because the classification model is created in a strange way, it is based on the detection model, deletes the detect head, and then replaces it with the classification head. The model is not created directly from yaml. This causes the classification model to fail when exported to tflite. I think this change is not difficult, but it is very useful. It can help users share YOLOv5 code modules and backbones to do image classification tasks.

ref: https://github.com/ultralytics/yolov5/blob/master/models/yolo.py (345-375)
https://github.com/ultralytics/yolov5/blob/master/export.py (1431) 

`assert not isinstance(model, ClassificationModel), "ClassificationModel export to TF formats not yet supported."`

`class ClassificationModel(BaseModel):
    """YOLOv5 classification model for image classification tasks, initialized with a config file or detection model."""

    def __init__(self, cfg=None, model=None, nc=1000, cutoff=10):
        """Initializes YOLOv5 model with config file `cfg`, input channels `ch`, number of classes `nc`, and `cuttoff`
        index.
        """
        super().__init__()
        self._from_detection_model(model, nc, cutoff) if model is not None else self._from_yaml(cfg)

    def _from_detection_model(self, model, nc=1000, cutoff=10):
        """Creates a classification model from a YOLOv5 detection model, slicing at `cutoff` and adding a classification
        layer.
        """
        if isinstance(model, DetectMultiBackend):
            model = model.model  # unwrap DetectMultiBackend
        model.model = model.model[:cutoff]  # backbone
        m = model.model[-1]  # last layer
        ch = m.conv.in_channels if hasattr(m, "conv") else m.cv1.conv.in_channels  # ch into module
        c = Classify(ch, nc)  # Classify()
        c.i, c.f, c.type = m.i, m.f, "models.common.Classify"  # index, from, type
        model.model[-1] = c  # replace
        self.model = model.model
        self.stride = model.stride
        self.save = []
        self.nc = nc

    def _from_yaml(self, cfg):
        """Creates a YOLOv5 classification model from a specified *.yaml configuration file."""
        self.model = None`

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jshh0401, thank you for bringing up this issue and for your detailed feedback on YOLOv5 🚀!

This is an automated response to let you know that an Ultralytics engineer will review your report and assist you soon.

If this is a 🐛 Bug Report, please provide a **minimum reproducible example (MRE)** if you haven't already. This helps us understand and resolve the problem efficiently.

For additional resources and guidance, you can refer to our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/), which cover a wide range of topics from [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) to [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

## Requirements
[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```
## Environments
YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your contribution and patience as we look into this! 🔎

**Comment by pderrenger** [comment]
Thanks for the detailed analysis @jshh0401! You're absolutely right that the current ClassificationModel implementation creates challenges for TFLite export since it's derived from detection models rather than built natively from YAML configs. Creating classification models directly from YAML would indeed be cleaner and resolve the TFLite export limitation. We'd welcome your PR to implement this enhancement - it would be a valuable contribution to make YOLOv5 classification workflows more robust across all export formats.

**Comment by jshh0401** [comment]
Thank you for your reply. I will work on this problem in the next few weekends. I plan to implement tflite export by adding TFClassify implementation in tf.py. If you have any suggestions, feel free to communicate with me.

**Comment by pderrenger** [comment]
That sounds like a solid approach @jshh0401! Adding TFClassify implementation in tf.py following the existing patterns for other TF export classes would be the right path. I'd suggest looking at how the existing TFDetect class handles the conversion as a reference, and ensure the TFClassify implementation properly handles the simplified architecture without the detection head. Feel free to open a draft PR when you start working on it so we can provide feedback during development.
